modules/renter.go

```diff
@@ -55,19 +55,18 @@ type ContractUtility struct {
 // DownloadInfo provides information about a file that has been requested for
 // download.
 type DownloadInfo struct {
-	SiaPath     string         `json:"siapath"`
-	Destination DownloadWriter `json:"destination"`
-	Filesize    uint64         `json:"filesize"`
-	Received    uint64         `json:"received"`
-	StartTime   time.Time      `json:"starttime"`
-	Error       string         `json:"error"`
-}
-
-// DownloadWriter provides an interface which all output writers have to implement.
-type DownloadWriter interface {
-	WriteAt(b []byte, off int64) (int, error)
-	Destination() string
-	Close() error
+	Destination     string `json:"destination"`     // The destination of the download.
+	DestinationType string `json:"destinationtype"` // Can be "file", "memory buffer", or "http stream".
+	Length          uint64 `json:"length"`          // The length requested for the download.
+	Offset          uint64 `json:"offset"`          // The offset within the siafile requested for the download.
+	SiaPath         string `json:"siapath"`         // The siapath of the file used for the download.
+
+	Completed           bool      `json:"completed"`           // Whether or not the download has completed.
+	EndTime             time.Time `json:"endtime"`             // The time when the download fully completed.
+	Error               string    `json:"error"`               // Will be the empty string unless there was an error.
+	Received            uint64    `json:"received"`            // Amount of data confirmed and decoded.
+	StartTime           time.Time `json:"starttime"`           // The time when the download was started.
+	TotalDataTransfered uint64    `json:"totaldatatransfered"` // The total amount of data transfered, including negotiation, overdrive, etc.
```

Comment:
what is "overdrive?" It doesn't appear to be documented in the download code either. If it's relevant to API consumers, it needs to be precisely defined in the docs

Reply:
I agree, it needs to be documented.

From Discord: I added an 'overdrive' feature to file downloads. It's not exposed in the api yet, but basically instead of downloading with 10 hosts, it'll download with 10+overdrive hosts. So if you set overdrive to 2, it'll download with 12 hosts.

Then, up to 2 of your hosts can be slow or fail and it won't impact the latency of the download
We experimented with this earlier, but with the previous download code any changes were pretty disruptive and unstable
now it's just an inherent part of downloading

Reply:
We can document it when we expose it via the api.